core/src/main/java/org/apache/calcite/rel/rules/PushProjector.java

```diff
@@ -250,6 +251,33 @@ public PushProjector(
         strongBitmap = ImmutableBitSet.range(nSysFields, nChildFields);
       }
 
+    } else if (childRel instanceof Correlate) {
+      Correlate joinRel = (Correlate) childRel;
```

Comment:
Better to rename the variable to correlateRel to distinguish it from the previous joinRel .